File: rust/tests/base_to_base_test.rs
use polyglot::base_to_base::base_to_base;

#[test]
fn test_base_to_base() {
    for (value, base, new_base, want) in TESTS {
        assert_eq!(base_to_base(value.to_string(), *base, *new_base), *want);
    }
}

const TESTS: &[(&'static str, u32, u32, &'static str)] = &[
    ("E", 16, 2, "1110"),
    (
        "11011110101011011011111011101111",
        2,
        3,
        "100122100210211112102",
    ),
    ("3735928559", 10, 4, "3132223123323233"),
    ("8831A383B", 12, 16, "DEADBEEF"),
];

File: typescript/tests/base_to_base_test.ts
import { assertEquals } from "https://deno.land/std/testing/asserts.ts";
import { baseToBase } from "../src/base_to_base.ts";

Deno.test("converts any base to another base", () => {
  for (const tc of TESTS) {
    assertEquals(baseToBase(tc.value, tc.base, tc.newBase), tc.want);
  }
});

const TESTS = [
  { value: "E", base: 16, newBase: 2, want: "1110" },
  {
    value: "11011110101011011011111011101111",
    base: 2,
    newBase: 3,
    want: "100122100210211112102",
  },
  { value: "3735928559", base: 10, newBase: 4, want: "3132223123323233" },
  { value: "8831A383B", base: 12, newBase: 16, want: "DEADBEEF" },
];
